src/services/dexes/uniswapV3/BarChartTicksGenerator.ts

```diff
@@ -0,0 +1,262 @@
+import {inject, injectable} from 'inversify';
+import {Token, CurrencyAmount} from '@uniswap/sdk-core';
+import {FeeAmount, Pool, TickMath, tickToPrice, Tick, TICK_SPACINGS} from '@uniswap/v3-sdk';
+import JSBI from 'jsbi';
+// eslint-disable-next-line max-len
+import IUniswapV3PoolABI from '@uniswap/v3-core/artifacts/contracts/interfaces/IUniswapV3Pool.sol/IUniswapV3Pool.json' assert {type: 'json'};
+
+import {ChainsIds} from '../../../types/ChainsIds.js';
+import {UniswapV3TickQuery} from './UniswapV3TickQuery.js';
+import {GraphTick, BarChartTick, TickProcessed} from './interfaces.js';
+import {BlockchainProviderRepository} from '../../../repositories/BlockchainProviderRepository.js';
+import {ethers} from 'ethers';
+
+enum Direction {
+  ASC,
+  DESC,
+}
+
+const MAX_INT128 = JSBI.subtract(JSBI.exponentiate(JSBI.BigInt(2), JSBI.BigInt(128)), JSBI.BigInt(1));
+
+@injectable()
+export class BarChartTicksGenerator {
+  @inject(BlockchainProviderRepository) blockchainProviderRepository!: BlockchainProviderRepository;
+  @inject(UniswapV3TickQuery) uniswapV3TickQuery!: UniswapV3TickQuery;
+
+  async apply(poolAddress: string, token0: Token, token1: Token, fee: FeeAmount, chainId: ChainsIds) {
+    const provider = this.blockchainProviderRepository.get(chainId);
+    const poolContract = new ethers.Contract(poolAddress, IUniswapV3PoolABI.abi, provider);
+    const [slot0, liquidity, graphTicks] = await Promise.all([
+      poolContract.slot0(),
+      poolContract.liquidity(),
+      this.uniswapV3TickQuery.apply(poolAddress),
+    ]);
+
+    const sdkTicks = graphTicks.map((graphTick: GraphTick) => {
+      return new Tick({
+        index: +graphTick.tickIdx,
+        liquidityGross: graphTick.liquidityGross,
+        liquidityNet: graphTick.liquidityNet,
+      });
+    });
+
+    const tickSpacing = TICK_SPACINGS[fee];
+    const fullPool = new Pool(token0, token1, fee, slot0.sqrtPriceX96, liquidity, slot0.tick, sdkTicks);
+    const activeTickIdx = Math.floor(fullPool.tickCurrent / tickSpacing) * tickSpacing;
+    const numSurroundingTicks = 100;
+
+    return await this.createBarChartTicks(
```

Comment:
I think this is the input method from uniswap documentation example, so let's create separate class for it and let's add commetn to the class with link to the example

Reply:
Agreed. Done!